File: kotlin/app/src/main/kotlin/com/hyperbabel/demo/api/UsersService.kt
/*
 * HyperBabel API — global block list endpoints. Blocks apply across rooms.
 */
package com.hyperbabel.demo.api

import kotlinx.serialization.json.JsonObject
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.HTTP
import retrofit2.http.POST
import retrofit2.http.Path

interface UsersService {
    @POST("api/v1/users/block")
    suspend fun blockUser(@Body body: Map<String, String>)

    @HTTP(method = "DELETE", path = "api/v1/users/block", hasBody = true)
    suspend fun unblockUser(@Body body: Map<String, String>)

    @GET("api/v1/users/{userId}/blocks")
    suspend fun getBlockList(@Path("userId") userId: String): JsonObject
}

File: swift/HyperBabelDemo/API/UsersService.swift
/*
 * HyperBabel API — global block list endpoints. Blocks apply across rooms.
 */
import Foundation

struct BlockedUser: Codable, Identifiable {
    var id: String { blockedId }
    let blockedId: String
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case blockedId = "blocked_id"
        case createdAt = "created_at"
    }
}

struct BlockListResponse: Codable {
    let blockedUsers: [BlockedUser]?

    enum CodingKeys: String, CodingKey {
        case blockedUsers = "blocked_users"
    }
}

enum UsersService {
    static func block(blockerId: String, blockedId: String) async throws -> EmptyResponse {
        try await ApiClient.shared.request(
            "POST", "/users/block",
            body: ["blocker_id": blockerId, "blocked_id": blockedId]
        )
    }

    static func unblock(blockerId: String, blockedId: String) async throws -> EmptyResponse {
        try await ApiClient.shared.request(
            "DELETE", "/users/block",
            body: ["blocker_id": blockerId, "blocked_id": blockedId]
        )
    }

    static func getBlockList(userId: String) async throws -> BlockListResponse {
        try await ApiClient.shared.request("GET", "/users/\(userId)/blocks")
    }
}
